service/src/test/java/bio/terra/workspace/amalgam/tps/TpsBasicTest.java

```diff
@@ -0,0 +1,149 @@
+package bio.terra.workspace.amalgam.tps;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.fail;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.delete;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.get;
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+
+import bio.terra.workspace.app.configuration.external.FeatureConfiguration;
+import bio.terra.workspace.common.BaseUnitTest;
+import bio.terra.workspace.generated.model.ApiTpsComponent;
+import bio.terra.workspace.generated.model.ApiTpsObjectType;
+import bio.terra.workspace.generated.model.ApiTpsPaoCreateRequest;
+import bio.terra.workspace.generated.model.ApiTpsPaoGetResult;
+import bio.terra.workspace.generated.model.ApiTpsPolicyInput;
+import bio.terra.workspace.generated.model.ApiTpsPolicyInputs;
+import bio.terra.workspace.generated.model.ApiTpsPolicyPair;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import java.util.UUID;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.MediaType;
+import org.springframework.mock.web.MockHttpServletResponse;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+import org.springframework.test.context.web.WebAppConfiguration;
+import org.springframework.test.web.servlet.MockMvc;
+import org.springframework.test.web.servlet.MvcResult;
+import org.springframework.test.web.servlet.setup.MockMvcBuilders;
+import org.springframework.web.context.WebApplicationContext;
+
+@ExtendWith(SpringExtension.class)
+// Shouldn't need since it is on the super class
+// @AutoConfigureMockMvc
+@WebAppConfiguration
+public class TpsBasicTest extends BaseUnitTest {
+  private static final String TERRA = "terra";
+  private static final String GROUP_CONSTRAINT = "group-constraint";
+  private static final String REGION_CONSTRAINT = "region-constraint";
+  private static final String GROUP = "group";
+  private static final String REGION = "region";
+  private static final String DDGROUP = "ddgroup";
+  private static final String US_REGION = "US";
+  @Autowired private ObjectMapper objectMapper;
+  @Autowired private WebApplicationContext webApplicationContext;
+  @Autowired private FeatureConfiguration features;
+  private MockMvc mockMvc;
+
+  @BeforeEach
+  public void setup() throws Exception {
+    this.mockMvc = MockMvcBuilders.webAppContextSetup(this.webApplicationContext).build();
+  }
```

Comment:
In other tests I've just used `@Autowired MockMvc mockMvc` (plus the `AutoConfigureMockMvc` annotation), does this setup do something different?

Reply:
I had a really hard time getting this test to work. I started with a vanilla MockMvc and that did not work. It did not do enough of the component scan to make TPS calls. I tried adding an explicit component scan which I also could not get to work. This form is what rusty, trusty Baeldung led me to...